src/main/java/dev/openfeature/sdk/OpenFeatureAPI.java

```diff
@@ -1,25 +1,35 @@
 package dev.openfeature.sdk;
 
-import java.util.*;
-import java.util.concurrent.ConcurrentHashMap;
-
-import javax.annotation.Nullable;
-
 import dev.openfeature.sdk.internal.AutoCloseableLock;
 import dev.openfeature.sdk.internal.AutoCloseableReentrantReadWriteLock;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.annotation.Nullable;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
 
 /**
  * A global singleton which holds base configuration for the OpenFeature library.
  * Configuration here will be shared across all {@link Client}s.
  */
+@Slf4j
 public class OpenFeatureAPI {
     // package-private multi-read/single-write lock
     static AutoCloseableReentrantReadWriteLock hooksLock = new AutoCloseableReentrantReadWriteLock();
     static AutoCloseableReentrantReadWriteLock contextLock = new AutoCloseableReentrantReadWriteLock();
-    private EvaluationContext evaluationContext;
+
     private final List<Hook> apiHooks;
-    private FeatureProvider defaultProvider = new NoOpProvider();
     private final Map<String, FeatureProvider> providers = new ConcurrentHashMap<>();
+    private final ExecutorService taskExecutor = Executors.newCachedThreadPool();
+    private final AtomicReference<FeatureProvider> initializingDefaultProvider = new AtomicReference<>();
+    private final Map<String, FeatureProvider> initializingNamedProviders = new ConcurrentHashMap<>();
```

Comment:
I think we need to clean up entries from this map 🤔

Reply:
🤦 regardless, of whether it would have a functional impact, this needs to be done. sorry, I missed that. will address this tonight. Thx for pointing out!

Reply:
Yeah, no functional impact. However, without the cleanup, we open up a memory leak. And depending on the provider's implementation details, there can be potential bugs :)